redux-form/lib/Field.d.ts

```diff
@@ -0,0 +1,321 @@
+
+    /**
+     * A Component, stateless function, or string corresponding to a default
+     * JSX element.
+     *
+     * Required but made optional so interface can be used on decorated components.
+     */
+    component?: ComponentConstructor | "input" | "select" | "textarea",
+    // component?: ComponentClass<P> | SFC<P> | "input" | "select" | "textarea",
+
+    /**
+     * If true, the rendered component will be available with the
+     * getRenderedComponent() method. Defaults to false. Cannot be used if your
+     * component is a stateless function component.
+     */
+    withRef?: boolean;
+
+    /**
+     * Formats the value from the Redux store to be displayed in the field input.
+     * Common use cases are to format Numbers into currencies or Dates into a
+     * localized date format.
+     */
+    format?: Formatter | null;
+
+    /**
+     * A function to convert whatever value the user has entered into the value that you
+     * want stored in the Redux store for the field.
+     *
+     * For instance, if you want the value to be in all uppercase, you would pass
+     *   value => value.toUpperCase()
+     */
+    normalize?: Normalizer;
+
+    /**
+     * Don't use.
+     */
+    props?: Object;
+
+    /**
+     * Parses the value given from the field input component to the type that you want
+     * stored in the Redux store. Common use cases are to parse currencies into Numbers into
+     * currencies or localized date formats into Dates.
+     */
+    parse?: Parser;
+
+    /**
+     * Allows you to to provide a field-level validation rule. The function will be given the
+     * current value of the field and all the other form values. If the field is valid, it
+     * should return `undefined`, if the field is invalid, it should return an error
+     * (usually, but not necessarily, a `String`).
+     */
+    validate?: Validator|Validator[];
+
+    /**
+     * Allows you to to provide a field-level warning rule. The function will be given the
+     * current value of the field and all the other form values. If the field needs a warning,
+     * it should return the warning (usually, but not necessarily, a `String`). If the field
+     * does not need a warning, it should return `undefined`.
+     */
+    warn?: Validator|Validator[];
+}
+
+interface Normalizer {
+    /**
+     * @param value             The value entered by the user.
+     * @param previousValue     The previous value for the field.
+     * @param allValues         All the values in the entire form with the new value.
+     *                          This will be an Immutable Map if you are using Immutable JS.
+     * @param previousAllValues All the values in the entire form before the current change.
+     *                          This will be an Immutable Map if you are using Immutable JS.
+     */
+    (value: FieldValue, previousValue?: FieldValue, allValues?: Object, previousAllValues?: Object): FieldValue;
+}
+
+interface Formatter {
```

Comment:
I believe so, you can provide formatter function in the config so you might want to have the type exported if for example this formatter is generated automatically in other function:
```typescript
function getFormatter (type: string): Formatter {
    // some logic to create formatter
}
```